### PR TITLE
Ki fulfillment

### PR DESCRIPTION
**Pull request by kiverso**
contains functionality for user stories 27 - 33, as well as 49 - 51. Users can view an order show page, an order index page, and cancel an order, canceling an order returns fulfilled items to the item's inventory. Merchants can fulfill items and fulfilling an item removes the ordered quantity from the inventory. When all items in an order are fulfilled the status changes to packaged. When the order is packaged it can be shipped my the admin. Admins can view all orders from their dashboard and packaged orders have a button to ship. Shipped orders can no longer be cancelled. The admin dashboard sorts orders by status, packaged first, followed by pending, followed by shipped, followed by cancelled.

### CODE REVIEW

File: app/controllers/admin/dashboard_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Admin::DashboardController < Admin::BaseController
+  def index
+    @orders = Order.order("CASE status WHEN 'packaged' THEN 0 WHEN 'pending'
```
Comment (danielpselph) [inline app/controllers/admin/dashboard_controller.rb:3]: Great job with the SQL injection

**Review by danielpselph — APPROVED** [review]
This is some really good work

File: app/controllers/merchant/base_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Merchant::BaseController < ApplicationController
+  before_action :require_merchant
+  
+  def require_merchant
```
Comment (arpariseau) [inline app/controllers/merchant/base_controller.rb]: This should have been in the latest application controller? We ought to pick one place or another.

File: app/controllers/profile/base_controller.rb [hunk 3]
```diff
@@ -1,7 +1,5 @@
 class Profile::BaseController < ApplicationController
-#  before:
-#   private
-#   def require_registered_user
-#     if 
-#   end
+  def require_registered_user
```
Comment (arpariseau) [inline app/controllers/profile/base_controller.rb:2]: same as with the merchant

**Review by arpariseau — APPROVED** [review]

**Pull request merged**